inotify.go

```diff
@@ -19,28 +20,39 @@ import (
 
 // Watcher watches a set of files, delivering events to a channel.
 type Watcher struct {
-	Events  chan Event
-	Errors  chan error
-	mu      sync.Mutex        // Map access
-	fd      int               // File descriptor (as returned by the inotify_init() syscall)
-	watches map[string]*watch // Map of inotify watches (key: path)
-	paths   map[int]string    // Map of watched paths (key: watch descriptor)
-	done    chan struct{}     // Channel for sending a "quit message" to the reader goroutine
+	Events   chan Event
+	Errors   chan error
+	mu       sync.Mutex // Map access
+	fd       int
+	poller   *fdPoller
+	watches  map[string]*watch // Map of inotify watches (key: path)
+	paths    map[int]string    // Map of watched paths (key: watch descriptor)
+	done     chan struct{}     // Channel for sending a "quit message" to the reader goroutine
+	doneresp chan struct{}     // Channel to respond to Close
```

Comment:
Maybe `doneResp` instead? Or `doneResposne`?

I was thinking of sending a signal one way over the done channel and then sending one back, but that won't work if the first signal is to close. Looking at the code more closely, I think we need to keep the two separate channels.